File: data/code_pairs/cpp/742.cpp
// clascl2.cpp
#include <iostream>
#include <complex>
#include <cassert>

using namespace std;

void clascl2(int m, int n, float* D, complex<float>* X, int ldx) {
    for (int j = 0; j < n; j++) {
        for (int i = 0; i < m; i++) {
            X[i + j*ldx] *= D[i];
        }
    }
}

int main() {
    const int M = 2, N = 3, LDX = 2;
    float D[M] = {2.0, 3.0};
    complex<float> X[LDX*N] = { {1.0,1.0}, {2.0,2.0}, {3.0,3.0},
                                {4.0,4.0}, {5.0,5.0}, {6.0,6.0} };

    clascl2(M, N, D, X, LDX);

    cout << "Scaled Matrix X:" << endl;
    for (int j = 0; j < N; j++) {
        for (int i = 0; i < M; i++) {
            auto val = X[i + j*LDX];
            cout << "(" << val.real() << "," << val.imag() << ") ";
        }
        cout << endl;
    }

    // Example assertion to verify correctness
    assert(X[0] == complex<float>(2.0, 2.0)); // The first element should be (2,2) after scaling

    cout << "Test completed successfully." << endl;

    return 0;
}

File: data/code_pairs/fortran/742.f90
! clascl2.f90
SUBROUTINE CLASCL2 ( M, N, D, X, LDX )
  INTEGER            M, N, LDX
  REAL               D( * )
  COMPLEX            X( LDX, * )
  INTEGER            I, J

  DO J = 1, N
     DO I = 1, M
        X( I, J ) = X( I, J ) * D( I )
     END DO
  END DO

END SUBROUTINE CLASCL2

PROGRAM test_clascl2
  IMPLICIT NONE
  INTEGER, PARAMETER :: M=2, N=3, LDX=2
  REAL :: D(M) = [2.0, 3.0]
  COMPLEX :: X(LDX,N) = RESHAPE([ (1.0,1.0), (2.0,2.0), (3.0,3.0), (4.0,4.0), (5.0,5.0), (6.0,6.0) ], [LDX,N])
  INTEGER :: i, j

  CALL CLASCL2(M, N, D, X, LDX)

  PRINT *, "Scaled Matrix X:"
  DO j = 1, N
     DO i = 1, M
        PRINT '(2F6.1)', REAL(X(i, j)), AIMAG(X(i, j))
     END DO
  END DO

END PROGRAM test_clascl2